pandas/core/reshape/reshape.py

```diff
@@ -162,6 +162,8 @@ def _make_selectors(self):
         self.full_shape = ngroups, stride
 
         selector = self.sorted_labels[-1] + stride * comp_index + self.lift
+        if np.prod(self.full_shape) > (2 ** 31 - 1):
+            raise ValueError('Pivot table is too big, causing int32 overflow')
```

Comment:
@jreback : Is it okay to catch it here, or should we try to catch earlier as you mentioned before?

Reply:
I'm happy to make any change, and would love to hear the reasoning for catching it earlier.

Reply:
ideally as soon as you know this is out of bounds you want to raise.